From 33313a656428b985d2c5a4eb3ab39b88b7628719 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Wed, 15 Jul 2026 18:21:38 -0400
Subject: [PATCH 190/268] win32u: Reset variables to zero within
 WM_WINE_WINDOW_STATE_CHANGED.

---
 dlls/win32u/message.c     | 4 ++--
 dlls/winex11.drv/window.c | 3 ---
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/dlls/win32u/message.c b/dlls/win32u/message.c
index 28eed562a35..73f27cdbb5b 100644
--- a/dlls/win32u/message.c
+++ b/dlls/win32u/message.c
@@ -2243,9 +2243,9 @@ static LRESULT handle_internal_message( HWND hwnd, UINT msg, WPARAM wparam, LPAR
     }
     case WM_WINE_WINDOW_STATE_CHANGED:
     {
-        UINT state_cmd, swp_flags;
+        UINT state_cmd = 0, swp_flags = 0;
         RECT window_rect;
-        HWND foreground;
+        HWND foreground = 0;
 
         if (!user_driver->pGetWindowStateUpdates( hwnd, &state_cmd, &swp_flags, &window_rect, &foreground )) return 0;
         window_rect = map_rect_raw_to_virt( window_rect, get_thread_dpi() );
diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index ce37a4c4d5e..10a635936a4 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -2064,9 +2064,6 @@ BOOL X11DRV_GetWindowStateUpdates( HWND hwnd, UINT *state_cmd, UINT *swp_flags,
     struct x11drv_win_data *data;
     HWND old_foreground;
 
-    *state_cmd = *swp_flags = 0;
-    *foreground = 0;
-
     if (!(old_foreground = NtUserGetForegroundWindow())) old_foreground = NtUserGetDesktopWindow();
     if (!is_virtual_desktop() && NtUserGetWindowThread( old_foreground, NULL ) == GetCurrentThreadId() &&
         !window_has_pending_wm_state( old_foreground, NormalState ) && !window_is_reparenting( old_foreground ) &&
-- 
2.55.0

